lib/daru/vector.rb

```diff
@@ -550,8 +549,14 @@ def sort opts={}, &block
         end
       } unless block
 
-      order = opts[:ascending] ? :ascending : :descending
-      vector, index = send(opts[:type], @data.to_a.dup, @index.to_a, order, &block)
+      block2 = lambda { |a, b|
+        block.call(a[1], b[1])
+      }
+
+      index_vector = @index.to_a.zip(@data)
+      index_vector = index_vector.sort &block2
```

Comment:
There seems no need to declare `block2` as a separate variable. Simply pass `{ |a, b| block.call(a[1], b[1]) }` to `#sort`.